components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -100,44 +104,75 @@ func (backend ES2Backend) getNodeReportIdsFromTimeseries(esIndex string,
 			"took",
 			"hits.total",
 			"aggregations.nodes.buckets.key",
-			"aggregations.nodes.buckets.distinct.hits.hits._id").
+			"aggregations.nodes.buckets.distinct.hits.hits._id",
+			"aggregations.nodes.buckets.distinct.hits.hits._source",
+		).
 		SearchSource(searchSource).
 		Do(context.Background())
 
 	if err != nil {
 		logrus.Errorf("unable to getNodeReportIdsFromTimeseries %v", err)
-		return []string{}, errors.Wrap(err, "getNodeReportIdsFromTimeseries unable to complete search")
+		return nil, errors.Wrap(err, "getNodeReportIdsFromTimeseries unable to complete search")
 	}
 
 	if searchResult.TotalHits() == 0 || searchResult.Hits.TotalHits == 0 {
 		logrus.Debugf("getNodeReportIdsFromTimeseries: No report ids for the given filters: %+v\n", filters)
 		// no matching report IDs is not an error, just return an empty array
-		return []string{}, nil
+		return nil, nil
 	}
+	LogQueryPartMin(esIndex, searchResult.Aggregations, fmt.Sprintf("%s searchResult aggs", myName))
 
 	outermostAgg, _ := searchResult.Aggregations.Terms("nodes")
 	if outermostAgg != nil {
 		for _, nodeBucket := range outermostAgg.Buckets {
 			topHits, _ := nodeBucket.Aggregations.TopHits("distinct")
 			nodeID := fmt.Sprintf("%s", nodeBucket.Key)
 			for _, hit := range topHits.Hits.Hits {
-				nodeReport[nodeID] = hit.Id
+				type endTimeSource struct {
+					EndTime time.Time `json:"end_time"`
+				}
+				var et endTimeSource
+				if hit.Source != nil {
+					err = json.Unmarshal(*hit.Source, &et)
+					if err == nil {
+
+					}
+				}
+				endTimeTimestamp, _ := ptypes.TimestampProto(et.EndTime)
+
+				repData := reportingapi.ReportData{}
+				repData.Id = hit.Id
+				repData.EndTime = endTimeTimestamp
+
+				nodeReport[nodeID] = repData
```

Comment:
since this function is now pushing 100 lines it might be nice to extract some chunks of it to private functions and/or move the type def out.

Reply:
I moved that typedef out